File: api-tester/src/main/java/com/revenuecat/apitester/java/PurchasesRuntimeExceptionAPI.java
package com.revenuecat.apitester.java;

import com.revenuecat.purchases.NoCoreLibraryDesugaringException;

@SuppressWarnings({"unused"})
final class PurchasesRuntimeExceptionAPI {
    static void check() {
        final String message = "message";
        final Throwable cause = new Throwable();
        throw new NoCoreLibraryDesugaringException(message, cause);
    }
}

File: api-tester/src/main/java/com/revenuecat/apitester/kotlin/PurchasesRuntimeExceptionAPI.kt
package com.revenuecat.apitester.kotlin

import com.revenuecat.purchases.NoCoreLibraryDesugaringException

@Suppress("unused", "UNUSED_VARIABLE", "ThrowingExceptionsWithoutMessageOrCause")
private class PurchasesRuntimeExceptionAPI {
    fun check() {
        val message = ""
        val cause = Throwable()
        val exception = NoCoreLibraryDesugaringException(message, cause)
        val runtimeException: RuntimeException = exception
        throw exception
    }
}
